she_codes_news/news/views.py

```diff
@@ -1,16 +1,81 @@
 from django.views import generic
-from .models import NewsStory
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse_lazy
+from .models import NewsStory, Category
+from .forms import StoryForm
+from django.contrib.auth.decorators import login_required
 
 
 class IndexView(generic.ListView):
     template_name = 'news/index.html'
+    context_object_name = 'all_stories'
 
     def get_queryset(self):
         '''Return all news stories.'''
-        return NewsStory.objects.all()
+        cat_id = self.request.GET.get("cat")
+        if cat_id:
+            return NewsStory.objects.filter(category=cat_id)
+        else:
+            return NewsStory.objects.all()
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['latest_stories'] = NewsStory.objects.all()[:4]
-        context['all_stories'] = NewsStory.objects.all()
+        context['categories'] = Category.objects.all()
```

Comment:
probably could have used a form here to avoid having to manually process the categories